src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -394,7 +387,8 @@ public string ToDisplayString(SymbolDisplayFormat format = null)
                 switch (IsNullable)
                 {
                     case true:
-                        if ((format.MiscellaneousOptions & SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier) != 0)
+                        if (IsAnnotated &&
```

Comment:
>IsAnnotated [](start = 28, length = 11)

You may want to take a look at  `AddNullableAnnotations` in the symbol display visitor. Maybe that code structure would be cleaner.
As it stands, I found it a bit confusing that we're checking `IsNullable` then checking `IsAnnotated`.